src/GAP.jl

```diff
@@ -105,8 +122,21 @@ function __init__()
     gap_module = @__MODULE__
     Base.MainInclude.eval(:(__JULIAGAPMODULE = $gap_module))
     if ! isdefined(Main, :__GAPINTERNAL_LOADED_FROM_GAP) || Main.__GAPINTERNAL_LOADED_FROM_GAP != true
-        run_it(GAPROOT, error_handler_func)
+        run_gap(GAPROOT, error_handler_func)
     end
+
+    ## FIXME: Hack because abstract types cannot be endowed with methods in 1.1
+    ## At 1.2, this should be possible and this hack can be replaced with the
+    ## correspponding line in ccalls.jl
+    MPtr = Base.MainInclude.eval(:(ForeignGAP.MPtr))
+    Base.MainInclude.eval(:(
+        (func::$MPtr)(args...; kwargs...) = GAP.call_gap_func(func, args...; kwargs...)
+    ))
 end
 
+include( "ccalls.jl" )
+include( "gap2.jl" )
```

Comment:
On the long run, I suggest we split up `gap2.jl` into multiple files with clearly defined content. Like `adapter.jl` containing `getindex`, `length` etc. methods (matching `adapter.gi`);  perhaps a `types.jl` with the definitions of `Obj`, `GAPInputType`, etc.

Reply:
Definitely, yes. I am also happy about a better name suggestion right now.